src/tir/transforms/compact_buffer_region.cc

```diff
@@ -452,11 +452,17 @@ class BufferCompactor : public StmtExprMutator {
 };
 
 PrimFunc CompactBufferAllocation(PrimFunc f) {
-  PrimFuncNode* fptr = f.CopyOnWrite();
-  std::unordered_map<Buffer, Region, ObjectPtrHash, ObjectPtrEqual> region =
-      BufferAccessRegionCollector::Collect(f);
-  fptr->body = BufferCompactor::Compact(f, region);
-  return f;
+  // Only apply this pass to TIR that is not from TE schedules
+  Optional<Bool> from_legacy_te_schedule = f->GetAttr("from_legacy_te_schedule", Bool(false));
+  if (!from_legacy_te_schedule.value()) {
+    PrimFuncNode* fptr = f.CopyOnWrite();
+    std::unordered_map<Buffer, Region, ObjectPtrHash, ObjectPtrEqual> region =
+        BufferAccessRegionCollector::Collect(f);
+    fptr->body = BufferCompactor::Compact(f, region);
+    return f;
+  } else {
+    return f;
+  }
```

Comment:
Shall we have a util function in `ir_utils.h` doing the check, like:

```
if (IsFromLegacyTeSchedule(f)) {
  // do nothing
  return f;
}
// do more with f
```

Reply:
Not sure what you mean @MasterJH5574.

I've implemented this suggestion.

Reply:
Got it. That's all right.